spring-cloud-stream-binder-kafka-test/src/test/java/org/springframework/cloud/stream/binder/kafka/KafkaTestBinder.java

```diff
@@ -32,22 +37,36 @@
  * @author Gary Russell
  * @author Soby Chacko
  */
-public class Kafka10TestBinder extends AbstractKafkaTestBinder {
+public class KafkaTestBinder extends AbstractKafkaTestBinder {
 
-	public Kafka10TestBinder(KafkaBinderConfigurationProperties binderConfiguration) {
+	@SuppressWarnings({ "rawtypes", "unchecked" })
+	public KafkaTestBinder(KafkaBinderConfigurationProperties binderConfiguration) {
 		try {
-			AdminUtilsOperation adminUtilsOperation = new Kafka10AdminUtilsOperation();
+			AdminUtilsOperation adminUtilsOperation = new Kafka11AdminUtilsOperation();
 			KafkaTopicProvisioner provisioningProvider =
 					new KafkaTopicProvisioner(binderConfiguration, adminUtilsOperation);
 			provisioningProvider.afterPropertiesSet();
 
-			KafkaMessageChannelBinder binder = new KafkaMessageChannelBinder(binderConfiguration, provisioningProvider);
+			KafkaMessageChannelBinder binder = new KafkaMessageChannelBinder(binderConfiguration,
+					provisioningProvider) {
+
+				/*
+				 * Some tests use multiple instance indexes for the same topic; we need to make
+				 * the error infrastructure beans unique.
```

Comment:
Do we have a story on the matter?

Reply:
Issue created: https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/issues/210